File: mixin-bekasi/src/main/java/com/andia/mixin/bekasi/visage/VisageColumn.java
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package com.andia.mixin.bekasi.visage;

import com.andia.mixin.Lean;
import com.andia.mixin.value.MixinColumn;
import com.andia.mixin.value.MixinColumnMetadata;
import com.andia.mixin.value.MixinType;
import com.fasterxml.jackson.annotation.JsonProperty;

public class VisageColumn extends Lean implements MixinColumn {

	@JsonProperty
	private Object key;
	private String type = MixinType.STRING.name();
	private VisageColumnMetadata metadata = new VisageColumnMetadata();

	public VisageColumn() {
		super(VisageColumn.class);
	}

	public VisageColumn(String key, String type) {
		this();
		this.key = key;
		this.type = type;
	}

	public VisageColumn(String key) {
		this(key, MixinType.STRING.name());
	}

	public VisageColumn(MixinColumn column) {
		this();
		this.key = column.getKey();
		this.type = column.getType();
		MixinColumnMetadata metadata = column.getMetadata();
		this.metadata = new VisageColumnMetadata(metadata);
	}

	@Override
	public Object getKey() {
		return key;
	}

	@Override
	public String getType() {
		return type;
	}

	@Override
	public MixinColumnMetadata getMetadata() {
		return metadata;
	}

}

File: mixin-bekasi-webface/src/main/js/bekasi/visage/VisageColumn.ts
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import Lean from "webface/core/Lean";

import { jsonLeanFactory } from "webface/constants";

import VisageColumnMetadata from "bekasi/visage/VisageColumnMetadata";

export default class VisageColumn extends Lean {

    public static LEAN_NAME = "VisageColumn";

    private key: string = null;
    private type: string = null;
    private metadata: VisageColumnMetadata = null;

    constructor(key?: string, type?: string) {
        super(VisageColumn.LEAN_NAME);
        this.setKey(key === undefined ? null : key);
        this.setType(type === undefined ? null : type);
    }

    public getKey(): any {
        return this.key;
    }

    public setKey(key: any): void {
        this.key = key;
    }

    public getType(): string {
        return this.type;
    }

    public setType(type: string): void {
        this.type = type;
    }

    public getMetadata(): VisageColumnMetadata {
        return this.metadata;
    }

}

jsonLeanFactory.register(VisageColumn.LEAN_NAME, VisageColumn);